### PR TITLE
enable test selection for default test handler

### LINKED ISSUE #53355: Selection of test classes / methods through run_test.py is broken
## 🐛 Bug

Selection of test classes / methods through `run_test.py` with `-i` is broken.

## To Reproduce

```sh
$ cd pytorch/test
$ TEST=test_dispatch.TestDispatch.test_def
$ python -m unittest $TEST
[...]
Ran 1 test in 12.792s
[...]
$ python run_test.py -i $TEST
[...]
Ran 28 tests in 40.443s
[...]
```

The actual test does not matter. I've just picked one at random.

## Expected behavior

Run only the selected test.

## Additional context

- This was broken in ~#43600~ #30445. Before we called `run_test` (the default `handler`) with `test_name`

    https://github.com/pytorch/pytorch/blob/e735395fc682cd59a44d04d52625a351b5374e6e/test/run_test.py#L434-L440

    whereas now we call with the `test_module`

    https://github.com/pytorch/pytorch/blob/854cc53594b4bb1871caad990e966e0de47abff8/test/run_test.py#L845-L850

- The only mention of this I could find is in #46272 and only because `test/distributed/test_distributed_*` cannot be run by `pytest` at the moment. Given that this is broken since ~half a year~ over a year, this is a testament that, although not officially supported yet, most of the developers might already be using `pytest`. Cc @rgommers @mruberry

### PR DESCRIPTION
**Pull request by pmeier**
Fixes #53355.

### CODE REVIEW

File: test/run_test.py [hunk 1]
```diff
@@ -941,12 +942,12 @@ def determine_target(test, touched_files, options):
...
 def run_test_module(test: str, test_directory: str, options) -> Optional[str]:
-    test_module = parse_test_module(test)
+    test_module, unittest_args = parse_test(test)
 
     # Printing the date here can help diagnose which tests are slow
     print_to_stderr('Running {} ... [{}]'.format(test, datetime.now()))
-    handler = CUSTOM_HANDLERS.get(test, run_test)
-    return_code = handler(test_module, test_directory, options)
+    handler = CUSTOM_HANDLERS.get(test_module, run_test)
+    return_code = handler(test_module, test_directory, options, extra_unittest_args=unittest_args)
```
Comment (pmeier) [inline test/run_test.py:950]: Currently, only `run_test`, the default `handler`, accepts the `extra_unittest_args` parameter. This shouldn't be that hard to add to the others

https://github.com/pytorch/pytorch/blob/d3cde6c23c1a7d184f38d4e5fb797257e39d3376/test/run_test.py#L603-L609

Do we want that?

Comment (mruberry) [reply test/run_test.py:950]: Let's see if anyone requests it.

**Comment by mruberry** [comment]
Looks like there are some tests failing.

**Comment by pmeier** [comment]
@mruberry Fixing this properly will take a lot more work than anticipated. Given that no one seems to use this functionality, I'm inclined to [agree with @rgommers](https://github.com/pytorch/pytorch/issues/53355#issuecomment-791921576):

> I'd say fix it anyway if it's easy to do so. If it takes more than 1-2 hours then never mind, given that it will become irrelevant once we manage to switch to pytest.

What is your opinion on this?

**Comment by mruberry** [comment]
> @mruberry Fixing this properly will take a lot more work than anticipated. Given that no one seems to use this functionality, I'm inclined to [agree with @rgommers](https://github.com/pytorch/pytorch/issues/53355#issuecomment-791921576):
> 
> > I'd say fix it anyway if it's easy to do so. If it takes more than 1-2 hours then never mind, given that it will become irrelevant once we manage to switch to pytest.
> 
> What is your opinion on this?

I also like to agree with @rgommers ;)

**Pull request closed without merging**